docs/src/main/sphinx/routines/examples.md

```diff
@@ -0,0 +1,371 @@
+# Example SQL routines
+
+
+After learning about [SQL routines from the
+introduction](/routines/introduction), the following sections show numerous
+examples of valid SQL routines. The routines are suitable as [inline
+routines](routine-inline) or [catalog routines](routine-catalog), after
+adjusting the name and adjusting the example invocations.
+
+The examples combine numerous supported statements. Refer to the specific
+statement documentation for further details:
+
+* [](/routines/function) for general SQL routine declaration
+* [](/routines/begin) and [](/routines/declare) for routine blocks
+* [](/routines/set) for assigning values to variables
+* [](/routines/return) for returning routine results
+* [](/routines/case) and [](/routines/if) for conditional flows
+* [](/routines/loop), [](/routines/repeat), and [](/routines/while) for looping constructs
+* [](/routines/iterate) and [](/routines/leave) for flow control
+
+A very simple routine that returns a static value without requiring any input:
+
+```sql
+FUNCTION answer()
+RETURNS BIGINT
+RETURN 42
+```
+
+## Inline and catalog routines
+
+A full example of this routine as inline routine and usage in a string
+concatenation with a cast:
+
+```sql
+WITH
+  FUNCTION answer()
+  RETURNS BIGINT
+  RETURN 42
+SELECT 'The answer is ' || CAST(answer() as varchar);
+-- The answer is 42
+```
+
+Provided the catalog `example` supports routine storage in the `default` schema,
+you can use the following:
+
+```sql
+USE example.default;
+CREATE FUNCTION example.default.answer()
+  RETURNS BIGINT
+  RETURN 42;
+```
+
+With the routine stored in the catalog, you can run the routine multiple times
+without repeated definition:
+
+```sql
+SELECT example.default.answer() + 1; -- 43
+SELECT 'The answer is' || CAST(example.default.answer() as varchar); -- The answer is 42
+```
+
+Alternatively, you can configure the SQL environment in the
+[](config-properties) to a catalog and schema that support SQL routine storage:
+
+```properties
+sql.default-function-catalog=example
+sql.default-function-schema=default
+```
+
+Now you can manage SQL routines without the full path:
+
+```sql
+CREATE FUNCTION answer()
+  RETURNS BIGINT
+  RETURN 42;
+```
+
+SQL routine invocation works without the full path:
+
+```sql
+SELECT answer() + 5; -- 47
+```
+
+## Declaration examples
+
+The result of calling the routine `answer()` is always identical, so you can
+declare it as deterministic, and add some other information:
```

Comment:
All functions are assumed to be deterministic by default.  If the function uses a non-deterministic function an error is raised, and the user must explicitly set the non deterministic flag.
```
trino:tpch> with function x() returns double return random()  select x();
Query 20231025_210845_00010_eiqt6 failed: line 1:1: Non-deterministic function declared DETERMINISTIC
```
```
with function x() returns double not deterministic return random()  select x();
-- 0.12345
```

Also the oppositie is true:
```
with function x() returns double not deterministic return 42  select x();
Query 20231025_211148_00013_eiqt6 failed: line 1:1: Deterministic function declared NOT DETERMINISTIC
```

Reply:
So should I say that they are all deterministitic by default?

Reply:
Added some info in function.md